src/pkg/term/colorizer.go

```diff
@@ -12,6 +12,54 @@ import (
 	"golang.org/x/term"
 )
 
+// DefaultBufferSize is the default number of recent messages to keep in the circular buffer
+var DefaultBufferSize = 10
+
+type circularBuffer struct {
+	size         int
+	linesWritten int
+	index        int
+	data         []string
+}
+
+func (c *circularBuffer) write(msg string) {
+	c.linesWritten++
+	if len(c.data) < c.size {
+		c.data = append(c.data, msg)
+	} else {
+		c.data[c.index] = msg
+	}
+	c.index = (c.index + 1) % c.size
+}
+
+func (c *circularBuffer) read() []string {
+	if c.linesWritten <= c.size {
+		return slices.Clone(c.data)
+	}
```

Comment:
I also wouldn't bother with this branch. I _think_ the logic you wrote below will handle this case just fine already. Write a few tests and convince yourself